### DISCUSSION TITLE
Tuple Validation in Pydantic

### DISCUSSION THREAD

**Discussion by telukwg** [pydantic/pydantic#9246] [Question]
Hi,
Is Tuple Validation possible in Pydantic?

I'm looking to create a list of tuples, where each tuple as a different type at each index position.  I'd like also to be able to specify a Pydantic UserModel for the individual tuple elements (perhaps RootModel is useful here?)  I notice it seems to be available in the JSON Schema spec - if I'm interpreting it correctly.

Regards, Bill.

https://json-schema.org/understanding-json-schema/reference/array#tupleValidation

**Comment by Viicos** [comment]
> Is Tuple Validation possible in Pydantic?

Yes, it is possible:

```python
from pydantic import BaseModel

class Model(BaseModel):
    f: list[tuple[int, str]]

Model(f=[(1, 1)])  # Validation error raised
```

You'll find plenty of info in the docs.
